plugins/continuous-toolbox/src/ContinuousToolbox.ts

```diff
@@ -0,0 +1,203 @@
+   */
+  protected convertToolboxItemToFlyoutItems(
+    toolboxItem: Blockly.IToolboxItem,
+  ): Blockly.utils.toolbox.FlyoutItemInfoArray {
+    let contents: Blockly.utils.toolbox.FlyoutItemInfoArray = [];
+    if (toolboxItem instanceof Blockly.ToolboxCategory) {
+      // Create a label node to go at the top of the category
+      contents.push({kind: 'LABEL', text: toolboxItem.getName()});
+      let itemContents = toolboxItem.getContents();
+
+      // Handle custom categories (e.g. variables and functions)
+      if (typeof itemContents === 'string') {
+        itemContents = [
+          {
+            custom: itemContents,
+            kind: 'CATEGORY',
+          },
+        ];
+      }
+      contents = contents.concat(itemContents);
+    }
+    return contents;
+  }
+
+  /**
+   * Updates the flyout's contents if it is visible.
+   */
+  override refreshSelection() {
+    if (this.getFlyout().isVisible()) {
+      if (this.refreshDebouncer) {
+        clearTimeout(this.refreshDebouncer);
+      }
+      this.refreshDebouncer = setTimeout(() => {
+        this.getFlyout().show(this.getInitialFlyoutContents());
+      }, 100);
+    }
+  }
+
+  /**
+   * Scrolls the flyout to display the newly selected category's contents.
+   *
+   * @param oldItem The previously selected toolbox category.
+   * @param newItem The newly selected toolbox category.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  override updateFlyout_(
+    oldItem: Blockly.ISelectableToolboxItem | null,
+    newItem: Blockly.ISelectableToolboxItem | null,
+  ) {
+    if (newItem) {
+      this.getFlyout().scrollToCategory(newItem);
+      if (!this.getFlyout().isVisible()) {
+        this.getFlyout().show(this.getInitialFlyoutContents());
+      }
+    } else if (this.getFlyout().autoClose) {
+      this.getFlyout().hide();
+    }
+  }
+
+  /**
+   * Returns whether or not the toolbox should deselect the old category.
+   *
+   * @param oldItem The previously selected toolbox category.
+   * @param newItem The newly selected toolbox category.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  override shouldDeselectItem_(
+    oldItem: Blockly.ISelectableToolboxItem | null,
+    newItem: Blockly.ISelectableToolboxItem | null,
+  ): boolean {
+    // Should not deselect if the same category is clicked again.
+    return !!(oldItem && oldItem !== newItem);
+  }
+
+  /**
+   * Gets a category by name.
+   *
+   * @param name Name of category to get.
+   * @returns Category, or null if not found.
+   * @internal
+   */
+  getCategoryByName(name: string): Blockly.ISelectableToolboxItem | null {
+    const category = this.getToolboxItems().find(
+      (item) =>
+        item instanceof Blockly.ToolboxCategory &&
+        item.isSelectable() &&
+        name === item.getName(),
+    );
+    if (category) {
+      return category as Blockly.ISelectableToolboxItem;
+    }
+    return null;
```

Comment:
Perhaps treat this as a guard?:
```suggestion
    if (!category) return null;
    return category as Blockly.ISelectableToolboxItem;
```

Reply:
Done.